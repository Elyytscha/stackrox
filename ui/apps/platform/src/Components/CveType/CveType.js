import React from 'react';
import PropTypes from 'prop-types';

import LabelChip from 'Components/LabelChip';

const cveTypes = ['IMAGE_CVE', 'K8S_CVE', 'ISTIO_CVE', 'NODE_CVE'];
const cveTypeMap = {
    IMAGE_CVE: 'Image CVE',
    K8S_CVE: 'Kubernetes CVE',
    ISTIO_CVE: 'Istio CVE',
    NODE_CVE: 'Node CVE',
};

const CveType = ({ types, context }) => {
    const sortedTypes = types.map((x) => cveTypeMap[x] || 'Unknown').sort();

    return context === 'callout' ? (
        <LabelChip type="base" text={`Type: ${sortedTypes.join(', ')}`} />
    ) : (
        <span>
            <div className="flex flex-col">
                {sortedTypes.map((cveType) => (
                    // eslint-disable-next-line react/jsx-key
                    <div className="flex justify-center" key={cveType}>
                        {cveType}
                    </div>
                ))}
            </div>
        </span>
    );
};

CveType.propTypes = {
    types: PropTypes.arrayOf(PropTypes.oneOf(cveTypes)),
    context: PropTypes.oneOf(['callout', 'bare']),
};

CveType.defaultProps = {
    types: [],
    context: 'bare',
};

export default CveType;
