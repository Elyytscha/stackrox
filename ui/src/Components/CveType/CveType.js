import React from 'react';
import PropTypes from 'prop-types';

import LabelChip from 'Components/LabelChip';

const cveTypes = ['IMAGE_CVE', 'K8S_CVE', 'ISTIO_CVE'];
const cveTypeMap = {
    IMAGE_CVE: 'Image CVE',
    K8S_CVE: 'Kubernetes CVE',
    ISTIO_CVE: 'Istio CVE',
};

const CveType = ({ type, context }) => {
    const typeText = cveTypeMap[type] || 'Unknown';

    return context === 'callout' ? (
        <LabelChip type="base" text={`Type: ${typeText}`} />
    ) : (
        <span>{typeText}</span>
    );
};

CveType.propTypes = {
    type: PropTypes.oneOf(cveTypes).isRequired,
    context: PropTypes.oneOf(['callout', 'bare']),
};

CveType.defaultProps = {
    context: 'bare',
};

export default CveType;
