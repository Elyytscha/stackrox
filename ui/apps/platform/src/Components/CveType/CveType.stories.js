import React from 'react';

import CveType from './CveType';

export default {
    title: 'CveType',
    component: CveType,
};

export const withDefaultStyle = () => (
    <div>
        <CveType types={['IMAGE_CVE']} />
        <h2 className="mt-8 mb-0">Text inherits its parent element properties</h2>
        <p>(example in a table cell)</p>
        <div className="rt-td w-1/10 p-2 flex items-center font-600 text-base-600 text-left border-r-0 leading-normal">
            <CveType types={['IMAGE_CVE']} />
        </div>
    </div>
);

export const withDefaultStyleK8sType = () => (
    <div>
        <CveType types={['K8S_CVE']} />
        <h2 className="mt-8 mb-0">Text inherits its parent element properties</h2>
        <p>(example in a table cell)</p>
        <div className="rt-td w-1/10 p-2 flex items-center font-600 text-base-600 text-left border-r-0 leading-normal">
            <CveType types={['K8S_CVE']} />
        </div>
    </div>
);

export const withDefaultStyleIstioType = () => (
    <div>
        <CveType types={['ISTIO_CVE']} />
        <h2 className="mt-8 mb-0">Text inherits its parent element properties</h2>
        <p>(example in a table cell)</p>
        <div className="rt-td w-1/10 p-2 flex items-center font-600 text-base-600 text-left border-r-0 leading-normal">
            <CveType types={['ISTIO_CVE']} />
        </div>
    </div>
);

export const withCalloutStyle = () => <CveType context="callout" types={['IMAGE_CVE']} />;

export const withCalloutStyleK8sType = () => <CveType context="callout" types={['K8S_CVE']} />;

export const withCalloutStyleIstioType = () => <CveType context="callout" types={['ISTIO_CVE']} />;
