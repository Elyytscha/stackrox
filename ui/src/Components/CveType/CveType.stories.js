import React from 'react';

import CveType from './CveType';

export default {
    title: 'CveType',
    component: CveType,
};

export const withDefaultStyle = () => (
    <div>
        <CveType type="IMAGE_CVE" />
        <h2 className="mt-8 mb-0">Text inherits its parent element properties</h2>
        <p>(example in a table cell)</p>
        <div className="rt-td w-1/10 p-2 flex items-center font-600 text-base-600 text-left border-r-0 leading-normal">
            <CveType type="IMAGE_CVE" />
        </div>
    </div>
);

export const withDefaultStyleK8sType = () => (
    <div>
        <CveType type="K8S_CVE" />
        <h2 className="mt-8 mb-0">Text inherits its parent element properties</h2>
        <p>(example in a table cell)</p>
        <div className="rt-td w-1/10 p-2 flex items-center font-600 text-base-600 text-left border-r-0 leading-normal">
            <CveType type="K8S_CVE" />
        </div>
    </div>
);

export const withDefaultStyleIstioType = () => (
    <div>
        <CveType type="ISTIO_CVE" />
        <h2 className="mt-8 mb-0">Text inherits its parent element properties</h2>
        <p>(example in a table cell)</p>
        <div className="rt-td w-1/10 p-2 flex items-center font-600 text-base-600 text-left border-r-0 leading-normal">
            <CveType type="ISTIO_CVE" />
        </div>
    </div>
);

export const withCalloutStyle = () => <CveType context="callout" type="IMAGE_CVE" />;

export const withCalloutStyleK8sType = () => <CveType context="callout" type="K8S_CVE" />;

export const withCalloutStyleIstioType = () => <CveType context="callout" type="ISTIO_CVE" />;
