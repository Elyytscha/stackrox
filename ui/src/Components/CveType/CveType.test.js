import React from 'react';
import { render } from '@testing-library/react';

import CveType from './CveType';

describe('CveType', () => {
    describe('default style', () => {
        test('should show an image CVE type in default style', async () => {
            const { asFragment } = render(<CveType type="IMAGE_CVE" />);

            expect(asFragment()).toMatchSnapshot();
        });

        test('should show a Kubernetes CVE type in default style', async () => {
            const { asFragment } = render(<CveType type="K8S_CVE" />);

            expect(asFragment()).toMatchSnapshot();
        });

        test('should show an Istio CVE type in default style', async () => {
            const { asFragment } = render(<CveType type="ISTIO_CVE" />);

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('callout style', () => {
        test('should show an image CVE type in callout style', async () => {
            const { asFragment } = render(<CveType context="callout" type="IMAGE_CVE" />);

            expect(asFragment()).toMatchSnapshot();
        });

        test('should show a Kubernetes CVE type in callout style', async () => {
            const { asFragment } = render(<CveType context="callout" type="K8S_CVE" />);

            expect(asFragment()).toMatchSnapshot();
        });

        test('should show an Istio CVE type in callout style', async () => {
            const { asFragment } = render(<CveType context="callout" type="ISTIO_CVE" />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
