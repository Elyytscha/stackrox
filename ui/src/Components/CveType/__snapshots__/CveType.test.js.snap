// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CveType callout style should show a Kubernetes CVE type in callout style 1`] = `
<DocumentFragment>
  <span
    class="inline-block border rounded font-600 text-center whitespace-no-wrap min-h-6 flex justify-center items-center text-base px-2 bg-base-200 border-base-400 text-base-800 capitalize "
    data-testid="label-chip"
  >
    <span>
      Type: Kubernetes CVE
    </span>
  </span>
</DocumentFragment>
`;

exports[`CveType callout style should show an Istio CVE type in callout style 1`] = `
<DocumentFragment>
  <span
    class="inline-block border rounded font-600 text-center whitespace-no-wrap min-h-6 flex justify-center items-center text-base px-2 bg-base-200 border-base-400 text-base-800 capitalize "
    data-testid="label-chip"
  >
    <span>
      Type: Istio CVE
    </span>
  </span>
</DocumentFragment>
`;

exports[`CveType callout style should show an image CVE type in callout style 1`] = `
<DocumentFragment>
  <span
    class="inline-block border rounded font-600 text-center whitespace-no-wrap min-h-6 flex justify-center items-center text-base px-2 bg-base-200 border-base-400 text-base-800 capitalize "
    data-testid="label-chip"
  >
    <span>
      Type: Image CVE
    </span>
  </span>
</DocumentFragment>
`;

exports[`CveType default style should show a Kubernetes CVE type in default style 1`] = `
<DocumentFragment>
  <span>
    Kubernetes CVE
  </span>
</DocumentFragment>
`;

exports[`CveType default style should show an Istio CVE type in default style 1`] = `
<DocumentFragment>
  <span>
    Istio CVE
  </span>
</DocumentFragment>
`;

exports[`CveType default style should show an image CVE type in default style 1`] = `
<DocumentFragment>
  <span>
    Image CVE
  </span>
</DocumentFragment>
`;
